src/main/java/com/esewa/usermanagement/service/AdminServiceImpl.java

```diff
@@ -0,0 +1,110 @@
+package com.esewa.usermanagement.service;
+
+import com.esewa.usermanagement.constants.ExceptionMessages;
+import com.esewa.usermanagement.entity.RegistrationLog;
+import com.esewa.usermanagement.entity.User;
+import com.esewa.usermanagement.enums.RoleType;
+import com.esewa.usermanagement.exceptions.UserRegistrationFailedException;
+import com.esewa.usermanagement.repository.UserRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+@Slf4j
+@Service
+public class AdminServiceImpl implements AdminService{
+
+    private final UserService userService;
+    private final UserRepository userRepository;
```

Comment:
I believe we only require userService in this class

Reply:
Done